lib/gias/importer.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+module Gias
+  class Importer < Service
+    def initialize(csv_path)
+      @csv_path = csv_path
+    end
+
+    attr_reader :csv_path
+
+    def call
+      Log.log('Gias::Importer', 'Importing GIAS schools...')
+
+      school_records = CSV.foreach(csv_path, headers: true).map(&:to_h)
+
+      GiasSchool.upsert_all(school_records, unique_by: :urn)
```

Comment:
I didn't understand this part.

* We expected a CSV, does the CSV will have only the new schools from a given timeframe?
* How many records we have as base to insert or update at one go? Should we check database locking?

Reply:
I read after is 2:30 a.m, so no need to concern with locking. yet